tests/integration/events.bats

```diff
@@ -107,3 +131,145 @@ function teardown() {
+  expect_stats_for_container "bb_two" "${lines[@]}"
+
+  teardown_running_container bb_one
+  teardown_running_container bb_two
+}
+
+@test "events --stats non-existent-container-1 non-existent-container-2 test_busybox (bails when any container doesn't exist)" {
+  # XXX: currently cgroups require root containers.
+  requires root
+
+  runc run -d --console-socket $CONSOLE_SOCKET test_busybox
+  [ "$status" -eq 0 ]
+
+  runc events --stats non-existent-container-1 non-existent-container-2 test_busybox
+  [ "$status" -ne 0 ]
+  [[ "${lines[0]}" =~ "non-existent-container-1" ]]
+  [[ "${lines[0]}" =~ "does not exist" ]]
+  [[ "${lines[1]}" =~ "non-existent-container-2" ]]
+  [[ "${lines[1]}" =~ "does not exist" ]]
+  [[ "${lines[@]}" != *"test_busybox"* ]]
+}
+
+@test "events --interval non-existent-container-1 non-existent-container-2 test_busybox (bails when any container doesn't exist)" {
+  # XXX: currently cgroups require root containers.
+  requires root
+
+  runc run -d --console-socket $CONSOLE_SOCKET test_busybox
+  [ "$status" -eq 0 ]
+
+  runc events --interval 100ms non-existent-container-1 non-existent-container-2 test_busybox
+  [ "$status" -ne 0 ]
+  [[ "${lines[0]}" =~ "non-existent-container-1" ]]
+  [[ "${lines[0]}" =~ "does not exist" ]]
+  [[ "${lines[1]}" =~ "non-existent-container-2" ]]
+  [[ "${lines[1]}" =~ "does not exist" ]]
+  [[ "${lines[@]}" != *"test_busybox"* ]]
+}
+
+@test "events --interval test_busybox test_busybox_disappearing (should exit when any container dies)" {
+  # XXX: currently cgroups require root containers.
+  requires root
+
+  runc run -d --console-socket $CONSOLE_SOCKET test_busybox
+  [ "$status" -eq 0 ]
+
+  runc run -d --console-socket $CONSOLE_SOCKET test_busybox_disappearing
+  [ "$status" -eq 0 ]
+
+  (__runc events --interval 100ms test_busybox test_busybox_disappearing > events.log && echo $? > runc_exitcode) &
+
+  retry 3 1 eval "grep -q 'test_busybox_disappearing' events.log"
+  teardown_running_container test_busybox_disappearing
+
+  wait
+
+  [ "$(cat runc_exitcode)" == "0" ]
+
+  [ -e events.log ]
+
+  run cat events.log
+  printf '%s\n' "${lines[@]}"
+}
+
+@test "events --interval test_busybox (should report oom events)" {
+  # XXX: currently cgroups require root containers.
+  requires root
+
+  runc run -d --console-socket $CONSOLE_SOCKET test_busybox
+  [ "$status" -eq 0 ]
+
+  # set memory limit to 10M
+  run eval "echo 10000000 > /sys/fs/cgroup/memory/test_busybox/memory.limit_in_bytes"
+  [ "$status" -eq 0 ]
+
+  (__runc events --interval 100ms test_busybox > events.log ) &
+
+  retry 3 1 eval "grep -q 'test_busybox' events.log"
+
+  # cause OOM
+  runc exec test_busybox dd if=/dev/urandom of=/dev/shm/too-big bs=1M count=12
+  [ "$status" -ne 0 ]
+
+  teardown_running_container test_busybox
+  wait
+
+  [ -e events.log ]
+
+  run cat events.log
+  printf '%s\n' "${lines[@]}"
+  expect_oom_for_container "test_busybox" "${lines[@]}"
+}
+
+function expect_stats_for_container() {
```

Comment:
Both of these functions can be implemented using grep AFAICS, which would be far more readable and wouldn't require having separate functions.